vllm/v1/worker/gpu_worker.py

```diff
@@ -327,6 +327,15 @@ def get_kv_cache_spec(self) -> dict[str, KVCacheSpec]:
     def initialize_from_config(self, kv_cache_config: KVCacheConfig) -> None:
         """Allocate GPU KV cache with the specified kv_cache_config."""
 
+        # Init kv cache connector here, because it requires
+        # `kv_cache_config`.
+        # NOTE(Kuntai): This need to be done before `initialize_kv_cache`,
+        # because `initialize_kv_cache` will inject kv cache groups not
+        # related to kv cache connector (e.g. kv cache sharing layers).
+        connector_vllm_config = copy.copy(self.vllm_config)
+        connector_vllm_config.kv_cache_config = copy.copy(kv_cache_config)
```

Comment:
We're using a copy of `VllmConfig` as a holder to send `KVCacheConfig` down to the connector? And `VllmConfig` doesn't ordinarily have a `kv_cache_config` member? That seems extremely brittle?

Why not just make `KVCacheConfig` a constructor parameter for `KVConnector`?

Or could we instead supply the connector the layer ID/name to KV cache group ID mapping? Will all connectors need this mapping to support HMA?